modules/global-classes/api.php

```diff
@@ -0,0 +1,204 @@
+		try {
+			$all = $this->repository->all();
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Getting global classes failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		return $all->get();
+	}
+
+	private function get( $request ) {
+		$id = $request->get_param( 'id' );
+
+		try {
+			$one = $this->repository->get( $id );
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Getting global class failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		if ( null === $one ) {
+			return new \WP_Error( 'entity_not_found', 'Global class not found', [ 'status' => 404 ] );
+		}
+
+		return $one;
+	}
+
+	private function delete( $request ) {
+		$id = $request->get_param( 'id' );
+
+		try {
+			$one = $this->repository->get( $id );
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Deleting global class failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		if ( null === $one ) {
+			return new \WP_Error( 'entity_not_found', 'Global class not found', [ 'status' => 404 ] );
+		}
+
+		try {
+			$this->repository->delete( $id );
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Deleting global class failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		return new \WP_REST_Response( null, 204 );
+	}
+
+	private function patch( \WP_REST_Request $request ) {
+		$id = $request->get_param( 'id' );
+		$updated = $request->get_params();
+
+		unset( $updated['id'] );
+
+		try {
+			$one = $this->repository->get( $id );
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Updating global class failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		if ( null === $one ) {
+			return new \WP_Error( 'entity_not_found', 'Global class not found', [ 'status' => 404 ] );
+		}
+
+		try {
+			$updated = $this->repository->patch( $id, $updated );
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Updating global class failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		return new \WP_REST_Response( $updated, 200 );
+	}
+
+	private function create( \WP_REST_Request $request ) {
+		$class = $request->get_params();
+
+		try {
+			$new = $this->repository->create( $class );
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Creating global class failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		return new \WP_REST_Response( $new, 201 );
+	}
+
+	private function arrange( $request ) {
+		$order = $request->get_params();
+
+		try {
+			$all = $this->repository->all();
+		} catch ( \Exception $e ) {
+			return new \WP_Error( 'unexpected_error', 'Arranging global classes failed unexpectedly', [ 'status' => 500 ] );
+		}
+
+		$differences = array_merge( $all->get_order()->diff( $order ), array_diff( $order, $all->get_order()->all() ) );
+
+		if ( ! empty( $differences ) ) {
+			return new \WP_Error( 'invalid_content', 'Global classes order invalid', [ 'status' => 422 ] );
+		}
```

Comment:
Maybe it should be in `validate_callback`?

Reply:
Why not using collection all the way? and seem the error message is not correct.
also I cant understand from the code what does it checks.

Reply:
There is a difference here, in my opinion, when the data is not in the format, we expect it's not valid, but when the the content is in the right format but wrong, it's invalid content. For example, if the ID the user sent us (when getting an ID) is in the right format, but we can't find the class we don't send validation error.

Reply:
I am checking if there are differences between the two arrays, I think it is understandable (I am using a variable called `differences`)